src/EuroDNS/Helper/EuroDNSApi.php

```diff
@@ -0,0 +1,1756 @@
+            $this->error = 'Something went wrong. Response is null on code. Please try again later.';
+            return '0';
+        }
+
+        // Load the XML response into the DOMDocument
+        $xmlDoc->loadXML($response);
+
+        // Get the 'result' element from the XML
+        $result = $xmlDoc->getElementsByTagName('result')[0];
+
+        if ($result) {
+            // Get the 'code' attribute from the 'result' element
+            $code = $result->getAttribute('code');
+
+            // Check if the code indicates an error
+            if ('1000' !== $code && '1001' !== $code) {
+                $msg = $xmlDoc->getElementsByTagName('msg');
+
+                // Iterate through each 'msg' element and append its text content to the error message
+                foreach ($msg as $desc) {
+                    $this->error .= $desc->textContent . "\n";
+                }
+
+                return $code;
+            }
+
+            return $code;
+        }
+
+        // Set an error message if the 'result' element is not found in the XML response
+        $this->error = 'Response is null on code.';
+
+        return '0';
+    }
+
+
+
+   /**
+     * Function to initiate domain transfer in EuroDNS.
+     *
+     * @param string $domainName - The domain name to be transferred.
+     * @param array $data - Additional data for the transfer.
+     *
+     * @return array - Result of the transfer initiation.
+     */
+    public function initiateTransfer($domainName, $data): array
+    {
+        // Set transfer parameters
+        $this->params = $data;
+
+        // Build XML request for domain transfer
+        $request = $this->generateDomainTransferRequest($domainName);
+
+        // Set the request property for reference
+        $this->request = $request;
+
+        // Connect to EuroDNS API to initiate the transfer
+        $response = $this->connect($request);
+
+        // Prepare the result array
+        $result = [];
+
+        // Check for errors and set the result accordingly
+        if (!empty($this->error)) {
+            $result['error'] = true;
+            $result['msg'] = $this->error;
+        } else {
+            $result['error'] = false;
+            $result['msg'] = $this->processResponse($response);
+        }
+
+        return $result;
+    }
+
+    /**
+     * This function is used to give nameserver details manually beacuse
+     *  upmind not providing any options to give nameserver deatils at transfer
+     */
+
+    function setManualNS($type,$domainName){
+
+        //check for there is any lookup nameservers available for this domain name
+        $nameserversArray = Utils::lookupNameservers($domainName);
+
+        $ns = null;
+
+        //if yes then generate xml with the  nameservers array
+        $i= 1;
+            // Loop through name servers and generate XML
+        foreach ($nameserversArray as $key => $val) {
+
+            $host = isset($val['host'])? $val['host']: $val;
```

Comment:
@ahaladh apologies for querying an accepted & merge PR, but wanted to make sure.

The `$nameserversArray` is expected to be an array of strings like

```
[
    'someone.ns.dns-server.com',
    'something.ns.dns-server.com'
]
```

So `$val` should never be an array to hold a `host` property. Unless I am missing something or you have a specific example?

Thanks